packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -1232,13 +1232,16 @@ abstract class ResidentRunner {
     }
   }
 
-  Future<bool> launchDevTools() async {
+  Future<bool> launchDevTools({bool openInBrowser = false}) async {
     if (!supportsServiceProtocol) {
       return false;
     }
     assert(supportsServiceProtocol);
     _devtoolsLauncher ??= DevtoolsLauncher.instance;
-    await _devtoolsLauncher.launch(flutterDevices.first.vmService.httpAddress);
+    await _devtoolsLauncher.launch(
```

Comment:
The following issue probably existed before as well, but if this `Future` does not complete, e.g. in this case if we never get the `Serving DevTools at ...` line or it is malformed, then flutter CLI terminal input will hang and the user will be stuck, afaict. We might just be able to update this to drop the `Future` instead of awaiting it, and check `_devToolsProcess` for `null` in `DevtoolsServerLauncher.close()`. @kenzieschmoll would you be able to check the behavior here by injecting a mistake into the `_serveDevToolsPattern` RegExp?